transportable-udfs-spark/src/main/scala/com/linkedin/transport/spark/StdUdfWrapper.scala

```diff
@@ -57,57 +57,68 @@ abstract class StdUdfWrapper(_expressions: Seq[Expression]) extends Expression
+            _stdUdf.asInstanceOf[StdUDF1[StdData, StdData]].getRequiredFiles(wrappedConstants(0))
+          case 2 =>
+            _stdUdf.asInstanceOf[StdUDF2[StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1))
+          case 3 =>
+            _stdUdf.asInstanceOf[StdUDF3[StdData, StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1), wrappedConstants(2))
+          case 4 =>
+            _stdUdf.asInstanceOf[StdUDF4[StdData, StdData, StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1), wrappedConstants(2), wrappedConstants(3))
+          case 5 =>
+            _stdUdf.asInstanceOf[StdUDF5[StdData, StdData, StdData, StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1), wrappedConstants(2), wrappedConstants(3), wrappedConstants(4))
+          case 6 =>
+            _stdUdf.asInstanceOf[StdUDF6[StdData, StdData, StdData, StdData, StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1), wrappedConstants(2), wrappedConstants(3), wrappedConstants(4), wrappedConstants(5))
+          case 7 =>
+            _stdUdf.asInstanceOf[StdUDF7[StdData, StdData, StdData, StdData, StdData, StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1), wrappedConstants(2), wrappedConstants(3), wrappedConstants(4), wrappedConstants(5), wrappedConstants(6))
+          case 8 =>
+            _stdUdf.asInstanceOf[StdUDF8[StdData, StdData, StdData, StdData, StdData, StdData, StdData, StdData, StdData]].getRequiredFiles(wrappedConstants(0),
+              wrappedConstants(1), wrappedConstants(2), wrappedConstants(3), wrappedConstants(4), wrappedConstants(5), wrappedConstants(6), wrappedConstants(7))
+          case _ =>
+            throw new UnsupportedOperationException("getRequiredFiles not yet supported for StdUDF" + _expressions.length)
         }
-      })
+
+        lazy val sparkContext = SparkSession.builder().getOrCreate().sparkContext
+        _distributedCacheFiles = requiredFiles.map(file => {
+          try {
+            val resolvedFile = FileSystemUtils.resolveLatest(file, FileSystemUtils.getHDFSFileSystem)
+            // TODO: Currently does not support adding of files with same file name. E.g dirA/file.txt dirB/file.txt
+            sparkContext.addFile(resolvedFile)
+            resolvedFile
+          } catch {
+            case e: IOException =>
+              throw new RuntimeException("Failed to resolve path: [" + file + "].", e)
+          }
+        })
+      }
     }
   } // scalastyle:on magic.number
 
-
-  private final def wrapConstants(): Seq[StdData] = {
-    _expressions.map(expr => {
-      if (expr.foldable) SparkWrapper.createStdData(expr.eval(), expr.dataType) else null
-    })
+  private final def checkNullsAndWrapConstants(): Array[StdData] = {
```

Comment:
Hive impl has wrapConstants in a separate method. Any reason why it is merged with null check? Can we have the same structure as Hive?

Reply:
To be consistent with checkNullsAndWrapArguments in Spark.
wrapConstants and wrapArguments are both separate methods in Hive because for DeferredObject it internally ensures that expression is only evaluated once. 
```
@Override
    public Object get() throws HiveException {
      if (!evaluated) {
        obj = eval.evaluate(rowObject, version);
        evaluated = true;
      }
      return obj;
    }
```
There is no such check in Spark's `Expression` and so I decided to combine checkNullsAndWrapArguments to avoid double evaluation. And I created  checkNullsAndWrapConstants to be consistent with checkNullsAndWrapArguments.